src/nats.h

```diff
@@ -2403,6 +2403,28 @@ natsConnection_Drain(natsConnection *nc);
 NATS_EXTERN natsStatus
 natsConnection_DrainTimeout(natsConnection *nc, int64_t timeout);
 
+/** \brief Signs any 'message' using the connection's user credentials.
+ *
+ * The connection must have been created with the #natsOptions_SetUserCredentialsFromFiles.
+ *
+ * This call will sign the given message with the private key extracted through
+ * the user credentials file.
+ * If there is no error and `*sig` is not `NULL`, the user must free the memory
+ * when no longer needed.
+ *
+ * @param nc the pointer to the #natsConnection object.
+ * @param message the byte array to sign.
+ * @param messageLen the length of the given byte array.
+ * @param sig the returned signature.
+ * @param sigLen the length of the returned signature.
+ */
+natsStatus
+natsConnection_Sign(natsConnection *nc,
+                    const unsigned char *message,
+                    int messageLen,
+                    unsigned char **sig,
```

Comment:
Do they malloc this pre-call or does the call?

Reply:
The call does allocate (because I do that at low level). So siglen is not required (see above).
I was thinking of having a version that does sign in a buffer, but the low level thing I had done does allocation, so was too much of a change for now.

Reply:
I don't think we should malloc the space for the sig.

Reply:
See above